modules/criteoBidAdapter.js

```diff
@@ -279,6 +291,7 @@ function buildCdbRequest(context, bidRequests, bidderRequest) {
       }
       if (bidRequest.params.nativeCallback || utils.deepAccess(bidRequest, `mediaTypes.${NATIVE}`)) {
         slot.native = true;
+        hasNativeSendId = hasNativeSendId || checkNativeSendId(bidRequest);
```

Comment:
I am not sure this ever will run the `checkNativeSendId` function?

On line 266 you are explicitly setting hasNativeSendId to true which means it will always short circuit in this OR operator and not run the `checkNativeSendId`.

Am I missing something?

![image](https://user-images.githubusercontent.com/13972794/79894410-413e9e80-83ba-11ea-8188-f90070add7bf.png)

Reply:
It's updated with improved tests, thanks for the pointing out.